test/integration/agent/flowexporter_test.go

```diff
@@ -177,3 +187,183 @@ func TestSetupConnTrackParameters(t *testing.T) {
+		if srcPort < baseSrcPort { // Overflow check
+			require.Fail(b, "Too many connections created")
+		}
+
+		// Create TCP connection
+		flow := conntrack.NewFlow(
+			6, // TCP protocol
+			0,
+			srcAddr,
+			dstAddr,
+			srcPort, // varying source port
+			80,      // destination port
+			3600,    // timeout
+			ctMark,  // mark - ConnSourceCTMarkField set to localVal
+		)
+		flow.Zone = ctZone
+
+		err := conn.Create(flow)
+		require.NoError(b, err, "Failed to create conntrack entry %d", i)
+		createdFlows = append(createdFlows, flow)
+	}
+
+	// Cleanup function to delete all created entries
+	cleanup := func() {
+		b.Logf("Cleaning up %d conntrack entries...", len(createdFlows))
+		for i, flow := range createdFlows {
+			if err := conn.Delete(flow); err != nil {
+				b.Logf("Warning: Failed to delete conntrack entry %d: %v", i, err)
+			}
+		}
+	}
+	defer cleanup()
+
+	// Create fake Kubernetes client and PodStore
+	var fakePods []runtime.Object
+
+	// Create source pod
+	srcPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:         "default",
+			Name:              "src-pod",
+			UID:               types.UID("src-pod-uid"),
+			CreationTimestamp: metav1.Time{Time: time.Now().Add(-time.Hour)},
+		},
+		Status: v1.PodStatus{
+			PodIPs: []v1.PodIP{
+				{IP: "10.0.0.2"},
+			},
+			Phase: v1.PodRunning,
+		},
+	}
+	fakePods = append(fakePods, srcPod)
+
+	// Create destination pods
+	for offset := range numDstIPs {
+		dstOctet := baseDstIPOctet + offset
+		dstIP := fmt.Sprintf("10.0.0.%d", dstOctet)
+		podName := fmt.Sprintf("dst-pod-%d", dstOctet)
+		podUID := fmt.Sprintf("dst-pod-%d-uid", dstOctet)
+
+		dstPod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:         "default",
+				Name:              podName,
+				UID:               types.UID(podUID),
+				CreationTimestamp: metav1.Time{Time: time.Now().Add(-time.Hour)},
+			},
+			Status: v1.PodStatus{
+				PodIPs: []v1.PodIP{
+					{IP: dstIP},
+				},
+				Phase: v1.PodRunning,
+			},
+		}
+		fakePods = append(fakePods, dstPod)
+	}
+
+	// Create fake client with all pods provided upfront
+	k8sClient := fake.NewSimpleClientset(fakePods...)
+
+	// Create PodStore with the fake client
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+
+	podInformer := coreinformers.NewPodInformer(k8sClient, metav1.NamespaceAll, 0, cache.Indexers{})
+	podInformer.SetTransform(k8s.NewTrimmer(k8s.TrimPod))
+	podStore := objectstore.NewPodStore(podInformer)
+
+	go podInformer.Run(stopCh)
+	cache.WaitForCacheSync(stopCh, podInformer.HasSynced, podStore.HasSynced)
+
+	// Create a real conntrack dumper that will read from the actual conntrack table
+	nodeConfig := &config.NodeConfig{
+		GatewayConfig: &config.GatewayConfig{
+			IPv4: net.ParseIP("10.0.0.1"),
+		},
+	}
+	serviceCIDRv4 := netip.MustParsePrefix("10.96.0.0/12")
+	connDumper := connections.NewConnTrackSystem(nodeConfig, serviceCIDRv4, netip.Prefix{}, false, filter.NewProtocolFilter(nil))
```

Comment:
I wonder how the benchmark fares on the different envs since we have different implementations between linux and "other"